treeherder/model/models.py

```diff
@@ -1137,6 +1137,70 @@ def __str__(self):
                                     self.build_system_type)
 
 
+class TextLogStep(models.Model):
+    '''
+    An individual step in the textual (unstructured) log
+    '''
+    id = BigAutoField(primary_key=True)
+
+    job = FlexibleForeignKey(Job)
+
+    # these are presently based off of buildbot results
+    # (and duplicated in treeherder/etl/buildbot.py)
+    SUCCESS = 0
+    TEST_FAILED = 1
+    BUSTED = 2
+    SKIPPED = 3
+    EXCEPTION = 4
+    RETRY = 5
+    USERCANCEL = 6
+    UNKNOWN = 7
+
+    RESULTS = ((SUCCESS, 'success'),
+               (TEST_FAILED, 'testfailed'),
+               (BUSTED, 'busted'),
+               (SKIPPED, 'skipped'),
+               (EXCEPTION, 'exception'),
+               (RETRY, 'retry'),
+               (USERCANCEL, 'usercancel'),
+               (UNKNOWN, 'unknown'))
+
+    name = models.CharField(max_length=200)
+    started = models.DateTimeField(null=True)
+    finished = models.DateTimeField(null=True)
+    started_line_number = models.PositiveIntegerField()
```

Comment:
If we store the byte offset instead of or in addition to the line number, we could facilitate clients sending HTTP Range requests to request subsets of logs. This allows clients to get the logs for a step without having to download the log and parse newlines.